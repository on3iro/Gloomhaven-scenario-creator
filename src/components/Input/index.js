/**
  * A simple styled controlled input element
  *
  * @namespace Input
  */

import React from 'react';
import styled from 'styled-components';


const StyledInput = styled.input`
  display: block;
  margin: 0 0 10px 0;
  padding: 5px 10px;
  border: 1px solid #e3e3e3;
  border-radius: 4px;
  color: #333;
  outline: 0;
`;

const Input = props => {
  return (
    <StyledInput {...props} />
  );
};

export default Input;
