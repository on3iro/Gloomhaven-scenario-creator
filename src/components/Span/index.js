import styled from 'styled-components';


const Span = styled.span`
  color: ${props => props.error ? 'red' : '#000'};
`;

export default Span;
