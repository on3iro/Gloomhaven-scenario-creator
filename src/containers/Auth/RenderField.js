import React from 'react';

import Input from 'components/Input';
import Span from 'components/Span';


const RenderField = ({ input, label, type, meta: { touched, error } }) => {
  return (
    <div>
      <Input
        {...input}
        placeholder={label}
        type={type}
        error={touched && error}
      />
      {touched && error && <Span error={touched && error}>{error}</Span>}
    </div>
  );
};

export default RenderField;
