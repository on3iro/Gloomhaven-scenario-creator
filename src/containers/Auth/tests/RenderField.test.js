import React from 'react';
import { shallow } from 'enzyme';

import Input from 'components/Input';
import Span from 'components/Span';

import RenderField from '../RenderField';


const renderComponent = newProps => {
  const props = {
    input: {},
    label: 'email',
    type: 'text',
    meta: {
      touched: false,
      error: false,
    },
    ...newProps,
  };

  const renderedWrapper = shallow(<RenderField {...props} />);

  return {
    props,
    renderedWrapper,
  };
};

describe('<RenderField />', () => {
  it('should render an <Input />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Input).length).toMatchSnapshot();
  });

  it('should render no <Span /> if it has not been touched or errored', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Span).length).toMatchSnapshot();
  });

  it('should render <Span /> when has been touched and errored', () => {
    const { renderedWrapper, props } = renderComponent({
      meta: {
        touched: true,
        error: true,
      }
    });
    expect(renderedWrapper.find(Span).length).toMatchSnapshot();
  });
});
