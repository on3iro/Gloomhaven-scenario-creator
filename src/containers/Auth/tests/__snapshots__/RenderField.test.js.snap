exports[`<RenderField /> should render <Span /> when has been touched and errored 1`] = `1`;

exports[`<RenderField /> should render an <Input /> 1`] = `1`;

exports[`<RenderField /> should render no <Span /> if it has not been touched or errored 1`] = `0`;
